sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_policies.py

```diff
@@ -10,14 +10,15 @@
 
 
 class CognitiveServicesCredentialPolicy(SansIOHTTPPolicy):
-    def __init__(self, cognitiveservices_key):
-        self.cognitiveservices_key = cognitiveservices_key
+    def __init__(self, shared_key_credential):
+        self.credential = shared_key_credential
         super(CognitiveServicesCredentialPolicy, self).__init__()
 
     def on_request(self, request):
+        subscription_key = self.credential._subscription_key  # pylint: disable=protected-access
```

Comment:
why not just make `_subscription_key` public if other things need to refer to it?

Reply:
The user instantiates a `SharedKeyCredential` and I don't really want them to know about subscription_key. If they want to update the key, they should use the available reset method. 

It shows up as the first suggestion in autocomplete/code completion in Pycharm without the underscore and I'd rather `set_subscription_key()` be what the user sees first.

Reply:
We should really avoid accessing private fields of other classes. Let's try to find another way to accomplish what you are looking for here.

Reply:
So, I may be off here (have only skimmed through the comment - but is there a reason that we have a private field and a public set method when we want to both read and write from the value? Why not make it a public attribute of the class?

Reply:
@johanste I think (@kristapratico please correct anything) that the motivation is to have a (documented) method for rotating this kind of credential. An alternative would be to put a similar method the `TextAnalyticsClient`, but then it would specialized and specific to only one sort of credential, which has its own downside (not being applicable to all kinds credentials a `TextAnalyticsClient` might have). *Personally* I would solve this by keeping `_subscription_key` private, and keeping the method (so that rotation is "a thing" in the API) but also provide a read-only `subscription_key` property. But I know read-only properties seem not to be favored here.

Reply:
Trying to follow the guidance here: https://github.com/Azure/azure-sdk/issues/947 which states to "not define public properties or fields which allow users to access the authentication data directly". 

Am I misunderstanding something? The user doesn't need read access to the value, I'm just reading it in the credential policy. The only way the user should interact with the credential is by `credential.set_subscription_key(key)` when they want to reset the key.

Reply:
@bryevdv I completely missed your comment when I submitted mine. That's exactly what I'm trying to communicate :)

Reply:
> Trying to follow the guidance here: Azure/azure-sdk#947 which states to "not define public properties or fields which allow users to access the authentication data directly".

ah, thanks for the link. I was looking for it earlier but for some reason didn't find it. The proposal there is to have a `refresh` method that takes (in this case) a subscription key. Which sounds reasonable as a general pattern across all credential types.

@bryevdv, I think that having a read-only property public property is all goodness in this case as there would very much be a potential for confusion to have the caller be able to directly set it (in this case, it would probably "just accidentally work"(tm), but in the general case - especially where there are multiple values that need to be changed - more interesting things may happen).